src/layouts/App/More.tsx

```diff
@@ -15,53 +20,131 @@ const ICON = styled(CenteredImg)`
+  text-align: start;
+  padding: 0px ${({ theme }) => theme.margin(2)};
+  height: 36px;
+  ${({ theme }) => theme.largeBorderRadius}
+  background-color: ${({ theme }) => theme.bg10};
+  border-color: ${({ theme }) => theme.bg10};
+  color: ${({ theme }) => theme.text4};
+  cursor: pointer;
+  margin-top: ${({ theme }) => theme.margin(0.5)};
 
-    > span {
-      color: ${({ theme }) => theme.text1};
-    }
+  ::placeholder {
+    color: #636363;
+    opacity: 1; /* Firefox */
   }
+  :-ms-input-placeholder {
+    /* Internet Explorer 10-11 */
+    color: #636363;
+  }
+  ::-ms-input-placeholder {
+    /* Microsoft Edge */
+    color: #636363;
+  }
+`
+
+const Button = styled.button`
+  width: 126px;
+  height: 40px;
+  margin: 13px 55px 0;
+  font-weight: 500;
+  font-size: 14px;
+  padding: 0 ${({ theme }) => theme.margin(2)};
+  ${({ theme }) => `padding-left: ${theme.margin(1.5)};`}
+  ${({ theme }) => theme.flexCenter};
+  border: none;
+  ${({ theme }) => theme.roundedBorders};
+  ${({ theme }) => theme.smallShadow};
+  background-color: ${({ theme }) => theme.secondary2};
+  cursor: pointer;
 `
-const SVGModeAdjust = styled.img`
-  filter: ${({theme}) => theme.filterWhiteIcon};
-`;
-
-const Overlay = ({ theme }: { theme: string }) => {
-  const target = {
-    discord: SOCIAL_MEDIAS.discord,
-    docs: LITEPAPER_ADDRESS,
-    medium: SOCIAL_MEDIAS.medium,
-    telegram: SOCIAL_MEDIAS.telegram,
-    twitter: SOCIAL_MEDIAS.twitter
-  } as { [key: string]: string }
+
+const Overlay = () => {
+  const [nodeURL, setNodeURL] = useState('')
+  const { setEndpoint } = useConnectionConfig()
+
+  const [rpcState, setRpcState] = useState({
+    endpoint: null,
+    endpointName: null,
+    network: null
+  })
+
+  const handleClickForRPC = (endpoint, endpointName, network) => {
+    setRpcState({ endpoint, endpointName, network })
+  }
+
+  const saveHandler = () => {
+    setEndpoint(rpcState.endpoint)
+    // analytics logger
+    const an = analytics()
+    an !== null && logEvent(an, 'rpc-selector', { ...rpcState })
+    notify({ message: `Switched to  ${rpcState.endpointName} (${rpcState.network}) ` })
+  }
 
   return (
-    <Menu>
-      {['docs', 'discord', 'twitter', 'telegram', 'medium'].map((item, index) => (
-        <ITEM key={index}>
-          <a href={target[item]} target="_blank" rel="noopener noreferrer">
-            <span>{item}</span>
-            <SVGModeAdjust src={`/img/assets/${item}_small.svg`}/> 
-          </a>
-        </ITEM>
-      ))}
-    </Menu>
+    <NewMenu>
+      <ITEM>
+        <ThemeToggle />
+      </ITEM>
+      <ITEM>
+        <ItemRow>
+          <span>Rpc EndPoint</span>
+          <SelectRPC handleClickForRPC={handleClickForRPC} />
+        </ItemRow>
+      </ITEM>
+      <ITEM>
+        <ItemRow>
+          <span>Node URL</span>
+          <INPUT
+            id="nodeURL"
+            onChange={(x: BaseSyntheticEvent) => {
+              !isNaN(x.target.value) && setNodeURL(x.target.value)
```

Comment:
The control statement `!isNaN(x.target.value)` is not asserting the correct case. The value of the input field will be a URL such as "https://ssc-dao.genesysgo.net". In this case, the value should never be a number. This function needs to set the context property `endpoint` with the value of the input field.

There are some additional requirements I will write up and send to you.